snmp/datadog_checks/snmp/data/profiles/hp-ilo4.yaml

```diff
@@ -0,0 +1,339 @@
+        name: cpqSm2NicRecvBytes
+      - OID: 1.3.6.1.4.1.232.9.2.5.2.1.10
+        name: cpqSm2NicRecvTotalPackets
+      - OID: 1.3.6.1.4.1.232.9.2.5.2.1.13
+        name: cpqSm2NicRecvDiscardPackets
+      - OID: 1.3.6.1.4.1.232.9.2.5.2.1.14
+        name: cpqSm2NicRecvErrorPackets
+      - OID: 1.3.6.1.4.1.232.9.2.5.2.1.15
+        name: cpqSm2NicRecvUnknownPackets
+    forced_type: monotonic_count # Sent as Counter (would be ingested as rates if not forced).
+    metric_tags:
+      - # Location of the controller (embedded controller vs. PC card).
+        # NOTE: other(1), embedded(2), pcmcia(3)
+        tag: nic_stats_location
+        column:
+          OID: 1.3.6.1.4.1.232.9.2.5.2.1.1
+          name: cpqSm2NicStatsLocation
+
+  - # Network physical adapters statistics (counts).
+    MIB: CPQNIC-MIB
+    table:
+      OID: 1.3.6.1.4.1.232.18.2.3.1
+      name: cpqNicIfPhysAdapterTable
+    symbols:
+      - # Number of frames successfully transmitted.
+        OID: 1.3.6.1.4.1.232.18.2.3.1.1.16
+        name: cpqNicIfPhysAdapterGoodTransmits
+      - # Number of frames successfully received.
+        OID: 1.3.6.1.4.1.232.18.2.3.1.1.17
+        name: cpqNicIfPhysAdapterGoodReceives
+      - # Number of frames not transmitted because of an error.
+        # NOTE: this is the sum of 4 other metrics. (As a result, we don't submit those 4 metrics.)
+        # NOTE: MIB recommends looking at details stats and take action if this counter increments frequently.
+        OID: 1.3.6.1.4.1.232.18.2.3.1.1.18
+        name: cpqNicIfPhysAdapterBadTransmits
+      - # Number of frames not received because of an error.
+        # NOTE: this is the sum of 4 other metrics. (As a result, we don't submit those 4 metrics.)
+        # NOTE: MIB recommends looking at details stats and take action if this counter increments frequently.
+        OID: 1.3.6.1.4.1.232.18.2.3.1.1.19
+        name: cpqNicIfPhysAdapterBadReceives
+      - # Number of octets received.
+        OID: 1.3.6.1.4.1.232.18.2.3.1.1.37
+        name: cpqNicIfPhysAdapterInOctets
+      - # Number of octets sent.
+        OID: 1.3.6.1.4.1.232.18.2.3.1.1.38
+        name: cpqNicIfPhysAdapterOutOctets
+    forced_type: monotonic_count  # Sent as Counters (not forcing would make us ingest them as rates).
+    metric_tags:
+      - tag: interface
+        column:
+          OID: 1.3.6.1.4.1.232.18.2.3.1.1.39
+          name: cpqNicIfPhysAdapterName
+
+  - # Network physical adapters statistics (rates).
+    MIB: CPQNIC-MIB
+    table:
+      OID: 1.3.6.1.4.1.232.18.2.3.1
+      name: cpqNicIfPhysAdapterTable
+    symbols:
+      - # Estimate of current bandwidth in bps.
+        # NOTE: this is zero if the value exceeds 4Gbps.
+        OID: 1.3.6.1.4.1.232.18.2.3.1.1.33
+        name: cpqNicIfPhysAdapterSpeed
+      - # Estimate of current bandwith in Mbps.
+        OID: 1.3.6.1.4.1.232.18.2.3.1.1.36
+        name: cpqNicIfPhysAdapterSpeedMbps
+    forced_type: counter  # Sent as Gauges
```

Comment:
Why a counter here? gauge seems to make more sense to me.

Reply:
Ah, so it's me being confused about {rate / counter} vs gauge whenever I encounter a "per second" metric. So we almost never submit rate metrics, do we?

Reply:
I feel that rate have a very narrow usage at least. Not exactly sure when they make the most sense.